metricflow/api/simple_api.py

```diff
@@ -0,0 +1,259 @@
+        """Retrieves a list of materialization names.
+
+        Returns:
+            A list of Materialization objects containing metadata.
+        """
+        return self.mf_engine.list_materializations()
+
+    def get_dimension_values(
+        self, metric_name: str, dimension_name: str, start_time: Optional[str] = None, end_time: Optional[str] = None
+    ) -> List[str]:
+        """Retrieves a list of dimension values given a [metric_name, dimension_name].
+
+        Args:
+            metric_name: Name of metric that contains the group_by.
+            dimension_name: Name of group_by to get values from.
+            start_time: Get data for the start of this time range.
+            end_time: Get data for the end of this time range.
+
+        Returns:
+            A list of dimension values as string.
+        """
+        parsed_start_time = convert_to_datetime(start_time)
+        parsed_end_time = convert_to_datetime(end_time)
+        return self.mf_engine.get_dimension_values(
+            metric_name=metric_name,
+            get_group_by_values=dimension_name,
+            time_constraint_start=parsed_start_time,
+            time_constraint_end=parsed_end_time,
+        )
+
+    def materialize(
+        self, materialization_name: str, start_time: Optional[str] = None, end_time: Optional[str] = None
+    ) -> SqlTable:
+        """Builds a table containing metrics and dimensions from a materialization definition.
+
+        This can be very expensive if a large time range is provided.
+
+        Args:
+            materialization_name: Name of materialization
+            start_time: Materialized for the start of this time range.
+            end_time: Materialized for the end of this time range.
+
+        Returns:
+            SqlTable object of the materialized table.
+        """
+        parsed_start_time = convert_to_datetime(start_time)
+        parsed_end_time = convert_to_datetime(end_time)
+        return self.mf_engine.materialize(
+            materialization_name=materialization_name,
+            time_constraint_start=parsed_start_time,
+            time_constraint_end=parsed_end_time,
+        )
+
+    def drop_materialization(self, materialization_name: str) -> bool:
+        """Drops the table associated with a materialization definition.
+
+        Args:
+            materialization_name: Name of materialization to drop.
+
+        Returns:
+            True if a table has been drop, False if table doesn't exist.
+        """
+        return self.mf_engine.drop_materialization(materialization_name=materialization_name)
+
+    def validate_configs(self) -> List[ValidationIssueType]:
+        """Validate a model according to configured rules.
+
+        Returns:
+            List of validation issues with the model provided.
+        """
+        issues = ModelValidator.validate_model(self.user_configured_model).issues
+        return [issue for issue in issues] if issues is not None else []
```

Comment:
Ok, this is weird looking but I realize it's doing a cast from the internal tuple to a list. There are two things here:

1. The `None` handling seems silly. Might be worth making the issues property a non-Optional `Tuple[ValidationIssueType, ...]` in a separate PR and then stacking this one back on top of it (or just merging it first and then updating this to trunk) - I can't think of a situation where we want to differentiate between None and tuple() over there.
2. To cast an Iterator to List, use `list()` instead of a list comprehension. It's easier for readers to understand, and according to The Internet it is a lot faster in the runtime.
```suggestion
        issues = ModelValidator.validate_model(self.user_configured_model).issues
        return list(issue) if issues is not None else []
```

Of course, if we change the typehints as suggested we can simply `return ModelValidator.validate_model(...).issues`

Reply:
I take it back on list() performance, other parts of The Internet say the comprehension is now as fast or faster. So do whatever. We should just use Tuple anyway.